fdsc.user.js

```diff
@@ -18,310 +18,400 @@
+                                    }
+                                }).done(function (data) {
+                                    // We use the first char of feedback to identify its type because that's what metasmoke does.
+                                    var tps = data.filter(function (el) { return el.feedback_type.indexOf('t') === 0; }).length;
+                                    var fps = data.filter(function (el) { return el.feedback_type.indexOf('f') === 0; }).length;
+                                    var naa = data.filter(function (el) { return el.feedback_type.indexOf('n') === 0; }).length;
+                                    if (tps == 0) {
+                                        $(".popup-actions").prepend("<div style='float:left' id='smokey-report'><strong>Smokey report: <span style='color:darkgreen'>" + tps + " tp</span>, <span style='color:red'>" + fps + " fp</span>, <span style='color:#7c5500'>" + naa + " naa</span></strong> - <a href='#' id='feedback-fp' style='color:rgba(255,0,0,0.5);' onMouseOver='this.style.color=\"rgba(255,0,0,1)\"' onMouseOut='this.style.color=\"rgba(255,0,0,0.5)\"'>false positive?</a></div>");
+                                    } else {
+                                        // If someone else has already marked as tp, you should mark it as fp in chat where you can discuss with others.
+                                        // Hence, do not display the false positive button
                                         $(".popup-actions").prepend("<div style='float:left' id='smokey-report'><strong>Smokey report: <span style='color:darkgreen'>" + tps + " tp</span>, <span style='color:red'>" + fps + " fp</span>, <span style='color:#7c5500'>" + naa + " naa</span></strong></div>");
-                                    })
-                                    .error(function(jqXHR, textStatus, errorThrown) {
-                                        StackExchange.helpers.showErrorMessage($(".topbar"), "An error occurred fetching post feedback from metasmoke.", {
-                                            'position': 'toast',
-                                            'transient': true,
-                                            'transientTimeout': 10000
-                                        });
-                                        console.log(jqXHR.status, jqXHR.responseText);
+                                    }
+                                    // On click of the false positive button
+                                    $("#feedback-fp").on("click", function (ev) {
+                                        console.log("Reporting as false positive");
+                                        ev.preventDefault()
+                                        if (!fdsc.msWriteToken || fdsc.msWriteToken === "null") {
+                                            fdsc.getWriteToken(false, function() { // Check that using false is correct?
```

Comment:
As long as the false positive button doesn't close the flag dialog, using `false` is correct. The `afterFlag` parameter should be set to `true` to make sure that a closing flag dialog doesn't also close the input box; if the flag dialog isn't going to be closing for a while at least, then using `false` is correct.